src/plugins/discover/public/application/view_components/utils/filter_columns.ts

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { IndexPattern } from '../../../opensearch_dashboards_services';
+
+/**
+ * Helper function to filter columns based on the fields of the index pattern.
+ * This function is used when we switch between index patterns. We want to keep the columns that are
+ * still available in the new index pattern and remove the ones that are not.
+ * If the resulting array is empty, it provides a fallback to a single '_source' column.
```

Comment:
We should use the default column from advanced settings and not hardcode `_source`. `DEFAULT_COLUMNS_SETTING`

Reply:
will update